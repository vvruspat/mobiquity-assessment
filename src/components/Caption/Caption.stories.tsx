import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Caption } from "./Caption";
import "../../index.css";

export default {
  title: "UI Kit/Typography/Caption",
  component: Caption,
} as ComponentMeta<typeof Caption>;

const Template: ComponentStory<typeof Caption> = (args) => (
  <Caption {...args}>Caption</Caption>
);

export const CaptionTemplate = Template.bind({});

CaptionTemplate.args = {};
