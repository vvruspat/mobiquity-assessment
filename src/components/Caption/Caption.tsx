import { FC, PropsWithChildren } from "react";
import { Text } from "../Text";

import s from "./Caption.module.css";

type CaptionProps = {};

/**
 * Subtitle component
 */
export const Caption: FC<PropsWithChildren<CaptionProps>> = ({ children }) => {
  return (
    <h6 className={s.caption}>
      <Text size="s" weight="regular">
        {children}
      </Text>
    </h6>
  );
};
