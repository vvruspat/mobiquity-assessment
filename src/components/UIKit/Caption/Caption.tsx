import {
  DetailedHTMLProps,
  FC,
  HTMLAttributes,
  PropsWithChildren,
} from "react";
import { Text } from "../Text";
import cn from "clsx";

import s from "./Caption.module.css";

type CaptionProps = DetailedHTMLProps<
  HTMLAttributes<HTMLHeadingElement>,
  HTMLHeadingElement
>;

/**
 * Subtitle component
 */
export const Caption: FC<PropsWithChildren<CaptionProps>> = ({
  children,
  className,
  ...htmlProps
}) => {
  return (
    <h6
      className={cn(s.caption, className)}
      {...htmlProps}
      data-testid="caption"
    >
      <Text size="s" weight="regular">
        {children}
      </Text>
    </h6>
  );
};
