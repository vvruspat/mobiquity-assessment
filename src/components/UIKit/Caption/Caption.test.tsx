import { render, screen } from "@testing-library/react";
import { Caption } from "./Caption";

test("renders caption", () => {
  render(<Caption>Test caption</Caption>);

  const el = screen.getByTestId("caption");

  expect(el).toHaveClass("caption");
  expect(el).toHaveTextContent("Test caption");
  expect(el).toMatchSnapshot();
});
