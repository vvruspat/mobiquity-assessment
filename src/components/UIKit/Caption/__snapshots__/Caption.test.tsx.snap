// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders caption 1`] = `
<h6
  class="caption"
  data-testid="caption"
>
  <span
    class="text size_s regular"
    data-testid="text"
  >
    Test caption
  </span>
</h6>
`;
